### PR TITLE
Add version 0.2 to `gitlab-set-status` task

### PR DESCRIPTION
**Pull request by rudeigerc**
/kind feature

# Changes

Add version 0.2 to the `gitlab-set-status` task.

This update introduced a `coverage` parameter for the `gitlab-set-status` task for displaying the coverage information in GitLab pipelines.

Reference: https://docs.gitlab.com/ee/api/commits.html#set-the-pipeline-status-of-a-commit

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/gitlab-set-status/0.2/gitlab-set-status.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    - name: STATE
+      description: |
+        The state of the status. Can be one of the following `pending`,
+        `running`, `success`, `failed`, or `canceled`.
+      type: string
+
+    - name: COVERAGE
+      description: |
+        The total code coverage.
+      type: string
```
Comment (vinamra28) [inline task/gitlab-set-status/0.2/gitlab-set-status.yaml:88]: just thinking if we should we make this optional for the users as not all might be posting the coverage. WDYT?

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

File: task/gitlab-set-status/0.2/gitlab-set-status.yaml (hunk 1 above)
Comment (rudeigerc) [reply task/gitlab-set-status/0.2/gitlab-set-status.yaml:88]: SGTM. Actually in my use cases, I trigger this task at the beginning (running) and the end (in `.finally`) of the pipeline.

Just set an empty string for it as the default.

**Pull request merged**